tests/cartesian_tests/integration_tests/multi_feature_tests/test_debug_backend.py

```diff
@@ -302,3 +302,26 @@ def test_stencil(
     test_stencil(field_in, field_out)
 
     np.testing.assert_allclose(field_out.view(np.ndarray)[:, :, :], 15)
+
+
+def test_k_only_access_stencil():
+    field_in = np.ones((4,), dtype=np.float64)
+    field_out = gt_storage.zeros(
+        dtype=np.float64, backend="debug", shape=(4, 4, 4), aligned_index=(0, 0, 0)
+    )
+    field_in[:] = [2, 3, 4, 5]
+
+    @gtscript.stencil(backend="debug")
+    def test_stencil(
+        in_field: gtscript.Field[gtscript.K, np.float64],
+        out_field: gtscript.Field[np.float64],
+    ):
+        with computation(PARALLEL):
+            with interval(0, 1):
+                out_field[0, 0, 0] = in_field[1]
```

Comment:
This truly should fail in my opinion. The default behavior that allows you to not index data dimensions is too magical. We have something showing in https://github.com/GEOS-ESM/NDSL/issues/43 that is around that behavior